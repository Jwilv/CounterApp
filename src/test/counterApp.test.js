import { shallow } from 'enzyme'
import React from 'react'
import CounterApp from '../counterApp';


describe('pruebas en counter app', () => {

  let wrapper = shallow(<CounterApp />)
  beforeEach( ()=>{
    wrapper = shallow(<CounterApp />)
  })

  test('pruebas de render en primera app', () => {

    const valor = 8;
    const wrapper = shallow(<CounterApp value={valor} />)
    expect(wrapper).toMatchSnapshot();
  });

  test('si no recibe argumento debe mostrar el valor por defecto', () => {
    const valorDefault = '0';

    const valor = wrapper.find('p').text()
    expect(valor).toBe(valorDefault);

  })

  test('simulacion btn1 (+1)', () => { 

    wrapper.find('button').at(0).simulate('click')
    const valorDefault = wrapper.find('p').text()
    expect(valorDefault).toBe('1')
   })
   test('simulacion btn3 (-1)', () => { 

    wrapper.find('button').at(2).simulate('click')
    const valorDefault = wrapper.find('p').text()
    expect(valorDefault).toBe('-1')

   })

   test('pruebas en el boton 2 (reset)', () => { 
    const reset = '10'
    const valor = 10;
    const wrapper = shallow(<CounterApp value={valor} />)
    wrapper.find('button').at(0).simulate('click')
 
    wrapper.find('button').at(1).simulate('click')
    const valorDefault = wrapper.find('p').text()

    expect(valorDefault).toBe(reset)

    })
});