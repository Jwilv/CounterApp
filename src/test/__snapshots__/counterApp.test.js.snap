// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en counter app pruebas de render en primera app 1`] = `
<Fragment>
  <h1>
    hola mundo
  </h1>
  <p>
    8
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
